OpenRA.Game/WorldUtils.cs

```diff
@@ -41,6 +41,94 @@ public static IEnumerable<Actor> FindActorsInCircle(this World world, WPos origi
 			}
 		}
 
+		/// <summary>
+		/// Finds all the actors of which their health radius is intersected by a line (with a definable width) between two points.
+		/// </summary>
+		/// <param name="world">The engine world the line intersection is to be done in.</param>
+		/// <param name="lineStart">The position the line should start at</param>
+		/// <param name="lineEnd">The position the line should end at</param>
+		/// <param name="lineWidth">How close an actor's health radius needs to be to the line to be considered 'intersected' by the line</param>
+		/// <returns>A list of all the actors 'intersected' by the line</returns>
+		public static IEnumerable<Actor> FindActorsOnLine(this World world, WPos lineStart, WPos lineEnd, WDist lineWidth, WDist targetExtraSearchRadius)
+		{
+			// This line intersection check is done by first just finding all actors within a square that starts at the source, and ends at the target.
+			// Then we iterate over this list, and find all actors for which their health radius is at least within lineWidth of the line.
+			// For actors without a health radius, we simply check their center point.
+			// The square in which we select all actors must be large enough to encompass the entire line's width.
+			var xMargin = Math.Sign(lineEnd.X - lineStart.X);
```

Comment:
These aren't margins. I suggesting calling them `xDir` and `yDir` to match how they are used below.